src/core/update.js

```diff
@@ -16,6 +16,9 @@ export let extractCss = (target) => {
  * @param {Object} sheet
  * @param {Boolean} append
  */
-export let update = (css, sheet, append) => {
-    sheet.data.indexOf(css) == -1 && (sheet.data = append ? css + sheet.data : sheet.data + css);
+export let update = (css, sheet, before) => {
+    sheet.data.indexOf(css) == -1 &&
+        (sheet.data = before
+            ? sheet.data.replace('/*' + before[0] + '*/', css + '/*' + before[0] + '*/')
```

Comment:
Alrighty, couldn't find time to look into it properly, but now I did. You did an excellent job in here! 😄  really love it!

Do you think we could actually use the bellow instead of adding a comment? 🤔 
```suggestion
            ? sheet.data.replace('.' + before[0], css + '.' + before[0])
```

If we do the above, the changes to `hash.js` are not needed, and that makes the size hit around `+10B` which is really awesome to fix this bug! 🎉

Reply:
Thanks for taking a look at this.
Well in some cases (like media queries) I think there might be some issues with your suggestion.
Like in this case 

```javascript
const RedBase = styled('div')`
  @media (max-width: 1245px) {
    color: #FF2222;
  }
  @media (max-width: 800px) {
    color: #FF4444;
  }
`;
```

In that case if you try to use `sheet.data.replace('.' + before[0], css + '.' + before[0])` it would be a problem:
```css
@media (max-width: 12450px) {
    /* THE NEW STYLE WOULD BE INSERTED HERE CAUSING A BUG */
    .go2087537148{color:#FF2222;}
}
@media (max-width: 800px) {
    .go2087537148{color:#FF2222;}
}
```

But adding comments before every style definitely feels hacky...
There's probably a better solution, it might be interesting to look at how styled-components or emotion handle this problem.

Reply:
Oh did not thought about media queries. Can you make me understand why that would bug out? 

Tried it with the changes in this codesandbox https://codesandbox.io/s/vanilla-forked-wprg1?file=/src/index.js:247-390 and to my untrained eye, looks fine? 🤔

Reply:
oh indeed my example does not work

I forked your codesandbox : [https://codesandbox.io/s/vanilla-forked-h06f6?file=/src/index.js](https://codesandbox.io/s/vanilla-forked-h06f6?file=/src/index.js)
You can see here that RedBase is black instead of red, because the class for red is inserted inside the first media-query for the orange style.

Reply:
Uh oh! Indeed!

So, that's happening because there are no other styles outside the media queries. Do you see this as an established pattern? To have a component solely based on media queries? 🤔